sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -936,8 +936,10 @@ def _raw_predict(self, X, n_threads=None):
             The raw predicted values.
         """
         is_binned = getattr(self, "_in_fit", False)
-        dtype = X_BINNED_DTYPE if is_binned else X_DTYPE
-        X = self._validate_data(X, dtype=dtype, force_all_finite=False, reset=False)
+        if not is_binned:
+            X = self._validate_data(
+                X, dtype=X_DTYPE, force_all_finite=False, reset=False
+            )
```

Comment:
I'm surprised that we have validation in this private method. Imo in the long term we should move the validation to the public methods but let's keep it that way in this PR